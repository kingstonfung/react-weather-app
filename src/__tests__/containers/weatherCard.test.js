import React from 'react';
import ReactDOM from 'react-dom';

import WeatherCard from '../../containers/weatherCard';

describe('WeatherCard component...', () => {
  const staticWeatherCard = (
    <WeatherCard
      city="somewhere"
      temperature="30"
      currentCondition="clear"
      cityImage="http://example.com/image.jpg"
      forecast={[]}
    />);

  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(staticWeatherCard, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const div = document.createElement('div');
    ReactDOM.render(staticWeatherCard, div);
    expect(div).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
  });
});