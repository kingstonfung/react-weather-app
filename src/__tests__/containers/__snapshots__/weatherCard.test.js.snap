// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard component... should match snapshot 1`] = `
<div>
  <div
    class="card "
  >
    <div
      class="cardUpperPortion"
      style="background-image: url(http://example.com/image.jpg);"
    >
      <div
        class="cardInfo"
      >
        <div
          class="cardTemperature"
        >
          30
        </div>
        <div
          class="currentCondition"
        >
          clear
        </div>
      </div>
      <h2
        class="cityNameTitle"
      >
        somewhere
      </h2>
    </div>
    <div
      class="cardBottomPortion"
    >
      <div
        class="cardDetail"
      >
        <ul
          class="forecastsWrapper"
        />
      </div>
    </div>
  </div>
</div>
`;
